src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,482 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+# Main entry point
+function gpkgread(fname::String; layer::Int=1, kwargs...)
+  db = SQLite.DB(fname)
+  
+  # 1. Read metadata to extract names of tables and CRS type
+  table_name, geom_column, crs_type = gpkgmetadata(db, layer)
+  
+  # 2. Get all column names from the table
+  columns_result = DBInterface.execute(db, "PRAGMA table_info(\"$table_name\")")
+  all_columns = [row.name for row in Tables.rows(columns_result)]
+  
+  # Filter out geometry column to get only attribute columns
+  attr_columns = filter(col -> col != geom_column, all_columns)
+  
+  # 3. Execute queries to get attribute and geometry data separately
+  if !isempty(attr_columns)
+    # Build query for attribute columns only
+    attr_columns_str = join(["\"$col\"" for col in attr_columns], ", ")
+    table_result = DBInterface.execute(db, "SELECT $attr_columns_str FROM \"$table_name\"")
+    # Materialize the query result into a NamedTuple
+    cols = Tables.columns(table_result)
+    names = Tables.columnnames(cols)
+    table = NamedTuple{names}([
+      Tables.getcolumn(cols, name) for name in names
+    ])
```

Comment:
As explained in a previous comment, there is no need to convert to named tuple. The `table_result` is already a valid table.